### DISCUSSION TITLE
How do I parse an html fragment?

### DISCUSSION THREAD

**Discussion by daniellizik** [syntax-tree/unist#69] [Q&A] [answered]
Say I have some arbitrary html such as

```html
parentless text node
<b>
  hello world
</b>
<p>
  <s>nested text node</s>
</p>
```
Seems that when I run this through rehype everything gets nested in `<html><body>....</body></html>`.

I can drill down into the AST to supply *just* the body node to the parser, but if I try to drill down into the children this ends up with

```
TypeError
Expected node, got `[object Object], ....`
```

React's `dangerouslySetInnerHTML` doesn't seem to be playing nicely with this, how can I parse just an array of nodes of type `element` in a fragment?

I have an example here https://codesandbox.io/s/purple-cache-flpie0?file=/src/index.js

**Comment by Murderlon** [comment]
Hi, this first of all seems like a [XY problem](https://meta.stackexchange.com/questions/66377/what-is-the-xy-problem/66378#66378). What is the problem instead of the attempted solution?

Your example also shows some custom logic to add styles but I would recommend using this: [rehype-document/#example-css](https://github.com/rehypejs/rehype-document/#example-css)

You also mention working with React, for which you could look at [react-markdown](https://github.com/remarkjs/react-markdown).

**Comment by daniellizik** [comment, marked as answer]
problem isn't related to this lib at all i found out
